types/src/vm/modules.rs

```diff
@@ -48,3 +40,11 @@ impl WithHash for AptosModuleExtension {
         &self.hash
     }
 }
+
+impl PartialEq for AptosModuleExtension {
+    fn eq(&self, other: &Self) -> bool {
+        self.hash.eq(&other.hash) && self.state_value_metadata().eq(other.state_value_metadata())
```

Comment:
why we skips the bytes in the new impl? optimization?

Reply:
We already have the hash, so should be faster